src/stdlib_math.fypp

```diff
@@ -1,21 +1,274 @@
 #:include "common.fypp"
 #:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
 
 module stdlib_math
     use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace
+    public :: EULERS_NUMBER_SP, EULERS_NUMBER_DP, EULERS_NUMBER_QP
+    public :: DEFAULT_LINSPACE_LENGTH, DEFAULT_LOGSPACE_BASE, DEFAULT_LOGSPACE_LENGTH
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    integer, parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace
+    real(sp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
```

Comment:
Is there a particular reason for hardcoding this number rather than evaluating it at compile time?
```suggestion
    real(sp), parameter :: EULERS_NUMBER_SP = exp(1.0_sp)
    real(dp), parameter :: EULERS_NUMBER_DP = exp(1.0_dp)
    real(qp), parameter :: EULERS_NUMBER_QP = exp(1.0_qp)
```

Reply:
No particular reason, I will incorporate your suggestion.